### PR TITLE
Integrate Cortex XDR incidents

### PR DESCRIPTION
**Pull request by paulmnguyen**
## Description

Support Cortex XDR Incidents in Palo Alto Networks Splunk App and Add-on.

This integration includes
- Splunk modinput to pull incidents via XDR API. (This integration does not use syslog)
- New 'Cortex XDR Incident' dashboard which replaces the previous 'All Incidents' dashboard.
- Splunk Common Information Model (CIM) compliant field extractions
- Splunk Enterprise Security (ES) notable events

## Motivation and Context

Traps has become Cortex XDR, so we will retire Traps support in the Splunk App and Add-on and replace it with Cortex XDR support.

## How Has This Been Tested?

Modular input tested with Cortex XDR API in Splunk 8.1. 
Incidents and notable events tested in Splunk ES.

## Types of changes

- New feature - Cortex XDR support
- Breaking change - removes Traps support

## Checklist

- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml [hunk 1]
```diff
@@ -0,0 +1,319 @@
+<form>
+  <label>Cortex XDR Incidents</label>
+  <search id="basesearch">
+    <query>sourcetype="pan:xdr_incident" $incident_id$ $severity$ $status$ | dedup incident_id | eval assigned_user_pretty_name=replace(assigned_user_pretty_name,"null", "UNASSIGNED") | eval score=coalesce(if(manual_score="null", null(), manual_score), if(rule_based_score="null", null(), rule_based_score))  | eval "Last Updated" = strftime(_time, "%Y-%d-%m %H:%M:%S") | spath  path="incident_sources{}" output="Incident Sources" | spath  path="hosts{}" output="dest_hosts" | table "Last Updated", starred, incident_id, description, assigned_user_pretty_name, severity, dest_hosts, score, status, "Incident Sources", xdr_url, "Open in XDR" </query>
```
Comment (btorresgil) [inline SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml:4]: 1. You're coalsceing some of these fields in props so shouldn't need to do it here.

2. Don't use dedup, it's too slow.

3. You're doing a lot in the base search that doesn't apply to all panels. Can you break this out a bit?

File: SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+        <option name="useThousandSeparators">1</option>
+      </single>
+    </panel>
+  </row>
+  <row>
+    <panel>
+      <title>Incident Severity Over Time</title>
+      <chart>
+        <search>
+          <query>sourcetype="pan:xdr_incident"  severity!=NULL | timechart  dc(incident_id) by severity</query>
```
Comment (btorresgil) [inline SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml:124]: Can this use the base search for performance optimization?

**Review by btorresgil — CHANGES_REQUESTED** [review]

File: Splunk_TA_paloalto/default/inputs.conf [hunk 3]
```diff
@@ -23,6 +23,16 @@ python.version = python3
...
 interval = 60
 
+#################
+# Cortex XDR inputs
+#################
+[cortex_xdr]
+index = default
+start_by_shell = false
+python.version = python3
+sourcetype = pan:xdr
```
[the remark this answers is not in the document]
Comment (paulmnguyen) [reply Splunk_TA_paloalto/default/inputs.conf:33]: Yes the modinput created this and isn't used. We plan to reuse this modular input so the generic sourcetype of pan:xdr is ok. The event injection will inject the proper sourcetype.

File: SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml (hunk 1 above)
Comment (paulmnguyen) [reply SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml]: Optimized the base search and dashboard.

File: SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml (hunk 2 above)
Comment (paulmnguyen) [reply SplunkforPaloAltoNetworks/default/data/ui/views/cortex_xdr_incidents.xml]: Using base search now

**Review by btorresgil — APPROVED** [review]

**Pull request merged**